File: FoodStock.Backend/src/FoodStock.Core/Entities/Organization.cs
namespace FoodStock.Core.Entities;

public class Organization
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string OwnerName { get; set; }
    public string OwnerSurname { get; set; }
    public string Nip { get; set; }
    public string Country  { get; set; }
    public string City { get; set; }
    public string CityCode { get; set; }
    public string Street { get; set; }
    public string StreetNumber { get; set; }

    public virtual List<Order> Orders { get; set; }
}

File: foodstock_mobile/lib/models/organization.dart
class Organization {
  final String id;
  final String name;
  final String ownerName;
  final String ownerSurname;
  final String nip;
  final String country;
  final String city;
  final String cityCode;
  final String street;
  final String streetNumber;

  Organization({
    this.id = "",
    required this.name,
    required this.ownerName,
    required this.ownerSurname,
    required this.nip,
    required this.country,
    required this.city,
    required this.cityCode,
    required this.street,
    required this.streetNumber,
  });

  factory Organization.fromJson(Map<String, dynamic> json) {
    return Organization(
      id: json['id'],
      name: json['name'],
      ownerName: json['ownerName'],
      ownerSurname: json['ownerSurname'],
      nip: json['nip'],
      country: json['country'],
      city: json['city'],
      cityCode: json['cityCode'],
      street: json['street'],
      streetNumber: json['streetNumber'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'ownerName': ownerName,
      'ownerSurname': ownerSurname,
      'nip': nip,
      'country': country,
      'city': city,
      'cityCode': cityCode,
      'street': street,
      'streetNumber': streetNumber,
    };
  }
}
